src/Uno.UI/UI/Xaml/Media/Animation/Animators/GPUFloatValueAnimator.iOSmacOS.cs

```diff
@@ -176,30 +213,26 @@ private void InitializeCoreAnimation()
 			}
 		}
 
-		public void Pause()
+		private void TrackCurrentInstance()
 		{
-			var pausedTime = _valueAnimator?.CurrentPlayTime;
-			var pausedValue = (float?)_valueAnimator?.AnimatedValue;
-
-			_valueAnimator?.Pause();
-			_coreAnimation?.Pause(pausedTime, pausedValue);
-
-			AnimationPause?.Invoke(this, EventArgs.Empty);
-		}
-
-		public void Resume()
-		{
-			_valueAnimator?.Resume();
-			_coreAnimation?.Resume();
+			lock(_weakActiveInstanceCache)
+			{
+				_weakActiveInstanceCache.Add(WeakReferencePool.RentWeakReference(this, this));
+			}
+			if (!_subscribedToVisibilityChanged &&
+				CoreWindow.GetForCurrentThread() is { } coreWindow)
+			{
+				coreWindow.VisibilityChanged += OnCoreWindowVisibilityChanged;
+				_subscribedToVisibilityChanged = true;
+			}
 		}
 
-		public void Cancel()
+		private void UntrackCurrentInstance()
 		{
-			_valueAnimator?.Cancel();
-			_coreAnimation?.Cancel();
-			AnimationCancel?.Invoke(this, EventArgs.Empty);
-
-			ReleaseCoreAnimation();
+			lock(_weakActiveInstanceCache)
+			{
+				_weakActiveInstanceCache.Remove(x => x.Target == this);
```

Comment:
To remove the target, you'll need to return the managed reference to the pool so it can be reused, same for the other location.